File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem54_SuccessfulPairsOfSpellsAndPotionsTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

class Problem54_SuccessfulPairsOfSpellsAndPotionsTest {

    private final Problem54_SuccessfulPairsOfSpellsAndPotions test = new Problem54_SuccessfulPairsOfSpellsAndPotions();

    @Test
    void successfulPairs() {
        int[] spells = new int[]{5, 1, 3};
        int[] potions = new int[]{1, 2, 3, 4, 5};

        int[] successfulPairs = new int[]{4, 0, 3};

        assertArrayEquals(successfulPairs, test.successfulPairs(spells, potions, 7));
    }
}

File: JavaScript/leetcode75/src/test/Problem54_SuccessfulPairsOfSpellsAndPotions.test.js
const successfulPairs = require("../main/Problem54_SuccessfulPairsOfSpellsAndPotions");

test("should return successful producst", () => {
  const spells = [5, 1, 3],
    potions = [1, 2, 3, 4, 5],
    success = 7;
  const product = [4, 0, 3];

  expect(successfulPairs(spells, potions, success)).toStrictEqual(product);
});
